src/net/sourceforge/kolmafia/Modifiers.java

```diff
@@ -1138,12 +1060,7 @@ public boolean setModifier(final Modifier mod) {
     }
 
     index -= BITMAP_MODIFIERS;
-    if (index < BOOLEAN_MODIFIERS) {
-      return this.setBoolean(index, mod.getValue().equals("true"));
-    }
-
-    index -= BOOLEAN_MODIFIERS;
-    return this.setString(index, mod.getValue());
+    return this.setBoolean(index, mod.getValue().equals("true"));
```

Comment:
I foresee a forthcoming PR addressing BooleanModifiers...

Reply:
going to do derived shortly, then I'm not sure. The complexity is that at the moment the implementation is very memory-efficient, in that it shoves all the booleans in the first bitmap modifier. The downside of doing that is that I don't really understand the code around it.

Reply:
bitmap modifiers are restricted to bits in a single int, aren't they?
Looks like we have 31 booleans.

Making code easier to understand & maintain is a plus.

You made a comment in another PR wondering why raveosity was a bitmap, but clownosity (built on clowniness) and surgeonosity are numeric. Considering that clowninessa can be 25, 50, or 75 - corresponding to Clownosity +1, +2, or +3 - and raveosity can be +1 or +2, it seems that bitmaps are obscure.

Perhaps it has something to do with Clowniness and Surgeonosity are the only two "Unique" modifiers - no more than one worn item that supplies them contributes to the score.

Investigating all that would be its own project,, I think.